plugins/com.python.pydev.analysis/src/com/python/pydev/analysis/visitors/OccurrencesVisitor.java

```diff
@@ -124,7 +125,7 @@ public OccurrencesVisitor(IPythonNature nature, String moduleName, IModule curre
     @Override
     public Object visitCompare(Compare node) throws Exception {
         Object ret = super.visitCompare(node);
-        if (isInTestScope == 0) {
+        if (isInTestScope == 0 && !(node.parent instanceof match_caseType)) {
```

Comment:
This is a bit odd...

Rather, a better way to do this would be complementing the `public void traverse(SimpleNode node) throws Exception {` to create a custom `traverse` for the `Match` statement incrementing the `isInTestScope` as needed (when visiting the `match.cases`).